src/middlewared/middlewared/plugins/system/info.py

```diff
@@ -192,14 +192,13 @@ def get_synced_clock_time(self):
         Will return synced clock time if ntpd has synced with ntp servers
         otherwise will return none
         """
-        client = ntplib.NTPClient()
         try:
-            response = client.request('localhost')
+            response = NTPClient('localhost').make_request()
         except Exception:
             # Cannot connect to NTP server
             self.logger.error('Error while connecting to NTP server', exc_info=True)
         else:
-            if response.version and response.leap != 3:
+            if response['version'] and response['leap'] != 3:
```

Comment:
Maybe we should return a (boolean) `valid` field from **inside** `format_ntp_packet` that encompasses this test

`response['version'] and response['leap'] != 3
`

Then we could use it both here, and in `test_ntp_server`

Reply:
Funny you see this, I'm actually looking at this logic. The link to a random ntp program actually has the inverse of our logic. So I believe this is actually a bug. (i.e. it should be `response['leap'] == 3` based on what I'm seeing). Still doing some research on this front.